### DISCUSSION TITLE
How to setup LanguageDetector with next-i18next

### DISCUSSION THREAD

**Discussion by aegiz** [i18next/next-i18next#1791] [Q&A] [answered]
Hi, I am trying to use LanguageDetector with next-i18next but I am getting the following error:

"Error: You are passing a wrong module! Please check the object you are passing to i18next.use()"

<img width="1319" alt="Screenshot 2022-04-12 at 20 09 58" src="https://user-images.githubusercontent.com/1919011/163026466-9b273a4e-2dc5-4079-90f5-d79d8fbc5cdb.png">

As seen in [the README](https://github.com/isaachinman/next-i18next#unserialisable-configs) there is a special configuration to have for this plugin to be serializable . So, I followed these steps and here are what my different files looks like:

```js
// next-i18next.config.js
const LanguageDetector = require('i18next-browser-languagedetector');
module.exports = {
    i18n: {
        defaultLocale: 'en',
        locales: ['en', 'jp'],
    },
    serializeConfig: false, // needed for LanguageDetector plugin to work
    use: [LanguageDetector],
};
```

```js
// next-config.js
const { i18n } = require('./next-i18next.config');
const moduleExports = {
    distDir: 'build',
    i18n,
    ...
 }
```

```ts
// _app.tsx
import nextI18NextConfig from '../next-i18next.config.js';
import { appWithTranslation } from 'next-i18next';
export default appWithTranslation(App, nextI18NextConfig);
```

```ts
// pages/index.tsx
import nextI18NextConfig from '../next-i18next.config.js';
export const getStaticProps: GetStaticProps = async ({ locale }) => {
    if (!locale) {
        return {
            props: {},
        };
    }
    return {
        props: await serverSideTranslations(locale, ['common'], nextI18NextConfig),
    };
};
```

Reading the [i18next-browser-languageDetector](https://github.com/i18next/i18next-browser-languageDetector) doc it seems like I need to call the `init` function. 
But where?

**Comment by isaachinman** [comment, marked as answer]
You do not need `i18next-browser-languagedetector` within a `next-i18next` project. NextJs itself handles all language detection.

**Reply by aegiz** [reply]
Duh! Makes sense. Thanks for your quick reply
